src/components/admin.tsx

```diff
@@ -789,12 +792,18 @@ function Admin({ user, isConductor = false }: adminProps) {
   };
 
   useEffect(() => {
-    checkTraceLangStatus(`${code}`).then((snapshot) =>
-      setTrackLanguages(snapshot.val())
-    );
-    checkTraceRaceStatus(`${code}`).then((snapshot) =>
-      setTrackRace(snapshot.val())
-    );
+    checkTraceLangStatus(`${code}`).then((snapshot) => {
+      setTrackLanguages(snapshot.val());
+      if (trackLanguages) {
```

Comment:
Is checking `trackLanguages` necessary ?
It look like you could check against the snapshot value itself, no ?

Reply:
Good idea, updated!